torch/_inductor/triton_ops/autotune.py

```diff
@@ -53,11 +54,14 @@ def __init__(self, fn, meta, configs, save_cache_hook, mutated_arg_names):
         self.configs = configs
         self.launchers = []
         self.lock = threading.Lock()
-        triton_cache_dir = os.path.join(
-            "/tmp", getpass.getuser(), str(self.meta.get("device", 0)), "triton/cache"
-        )
-        os.environ["TRITON_CACHE_DIR"] = triton_cache_dir
-        log.info(f"Triton cache directory: {triton_cache_dir}")
+        if os.getenv("TRITON_CACHE_DIR") is None:
```

Comment:
Actually, this might be the wrong place to set this value -- there are some places where we don't actually use CachingAutotuner

Reply:
Oh this is a good point.  Do we always have inductor decorators (@pointwise, etc.) on the triton kernels we generate?  If not, they'll only go through Triton's tuner, right?

Reply:
Actually, that won't get it either.  The other place we call Triton is here:
https://github.com/pytorch/pytorch/blob/b399007a073d3a9ba4f6668c00d368fc9ee0f59a/torch/_inductor/select_algorithm.py#L383-L385

Reply:
@jansel I think the main problem I had with PyCodeCache I may not be able to get device idx. But it seems I can call 

```
cuda_properties. current_device
```

So can we conclude that we can make it inside PyCodeCache?

Reply:
@jansel I think another problem with setting this in TritonCodeCache or PyCodeCache is that it's a little higher level (called from getting fx graph). So if we already have a triton codegen it'll be called here. I think it's better to call it in lower level